verilog/analysis/symbol_table.h

```diff
@@ -131,6 +131,15 @@ struct ReferenceComponent {
   const SymbolTableNode* resolved_symbol = nullptr;
 
  public:
+  ReferenceComponent(
```

Comment:
If we just remove he constructors below, we should not need this workaround (the constructors below are just default copy and move)